### PR TITLE
Replace emptyArray with []

### PR DESCRIPTION
**Pull request by jakebailey**
Redux of #49269, but with tests passing.

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot perf test this

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -7178,7 +7178,8 @@ namespace ts {
         if (!diagnostic.relatedInformation) {
             diagnostic.relatedInformation = [];
         }
-        Debug.assert(diagnostic.relatedInformation !== emptyArray, "Diagnostic had empty array singleton for related info, but is still being constructed!");
+        // TODO(jakebailey): what to do?
+        // Debug.assert(diagnostic.relatedInformation !== [], "Diagnostic had empty array singleton for related info, but is still being constructed!");
```
Comment (jakebailey) [inline src/compiler/utilities.ts:7182]: This is unfortunate.

**Comment by jakebailey** [comment]
@typescript-bot perf test this and i'm sorry I broke your previous run

**Comment by jakebailey** [comment]
Hm, checking is faster, but binding is slower? Or it's noise?

**Pull request closed without merging**